app/src/main/java/de/lilithwittmann/voicepitchanalyzer/utils/RecordingCleaner.java

```diff
@@ -0,0 +1,117 @@
+        cleanUnfinishedRecordings(this.context);
+        cleanRecordings(this.context);
+    }
+
+    public static void cleanUnfinishedRecordings(Context context) {
+        Path directory = RecordingPaths.getUnfinishedRecordingsDirectoryPath(context);
+        if (directory == null || !Files.isDirectory(directory)) {
+            return;
+        }
+        DirectoryStream<Path> directoryStream;
+        try {
+            directoryStream = Files.newDirectoryStream(directory);
+        } catch (IOException ex) {
+            Log.w(LOG_TAG, "error opening recording directory " + directory + ": " + ex);
+            return;
+        }
+
+        long deleteCount = 0;
+        for (Path filePath : directoryStream) {
+            try {
+                Files.deleteIfExists(filePath);
+                deleteCount += 1;
+            } catch (IOException ex) {
+                Log.w(LOG_TAG, "error deleting unfinished recording " + filePath + ": " + ex);
+            }
+        }
+
+        Log.i(LOG_TAG, "deleted " + deleteCount + " unfinished recording files");
+    }
+
+    public static void cleanRecordings(Context context) {
+        RecordingDB db = new RecordingDB(context);
+
+        Date newestDeleteDate = Date.from(Instant.now().minus(Duration.ofDays(MIN_RECORDING_DAYS)));
+        long runningSize = 0;
+        long deletedSize = 0;
+        long runningCount = 0;
+        long deleteCount = 0;
+        long oldCount = 0;
+        for (Recording recording : db.getRecordingsWithFiles()) {
+            runningCount += 1;
+            runningSize += recording.getRecordingFileSize();
+            if (recording.getDate().before(newestDeleteDate)) {
+                oldCount += 1;
+                if (runningSize > MAX_RECORDINGS_SIZE) {
+                    DeleteResult deleteResult = deleteRecording(context, db, recording);
+                    if (deleteResult == DeleteResult.OK || deleteResult == DeleteResult.MISSING) {
+                        deleteCount += 1;
+                    }
+                    if (deleteResult == DeleteResult.OK) {
+                        deletedSize += recording.getRecordingFileSize();
+                    }
+                }
+            }
+        }
+
+        Log.i(LOG_TAG, "deleted " + deleteCount + " of " + oldCount + " old recording files (of " + runningCount + " total) freeing " + deletedSize
+              + " of " + runningSize + " bytes");
+    }
+
+    private static DeleteResult deleteRecording(Context context, RecordingDB db, Recording recording) {
+        Path path = RecordingPaths.getRecordingPath(context, recording.getRecording());
+        if (path == null) {
+            return DeleteResult.ERROR;
+        }
+        DeleteResult result;
+        try {
+            Files.delete(path);
+            result = DeleteResult.OK;
+        } catch (NoSuchFileException ex) {
+            Log.w(LOG_TAG, "recording file " + path + " missing");
+            result = DeleteResult.MISSING;
```

Comment:
We'll update the filename NULL if the file isn't found - I guess that's reasonable.